components/screens/MainScreen.tsx

```diff
@@ -33,43 +40,61 @@ class MainScreen extends React.Component<Props, State> {
    * @param props props
    */
   constructor(props: Props) {
+
     super(props);
     this.state = {
+      loading : false,
+      quizData : [],
+      optionsArray : [],
     };
   }
 
   /**
    * Navigation options
    */
-  static navigationOptions = (props: HeaderProps) => {
-    return ({ 
-      headerTitle: <TopBar navigation={props.navigation} showMenu={true} showHeader={false} showLogout={true} showUser={true} />
+  public static navigationOptions = (props: HeaderProps) => {
+    return ({
+      headerTitle: <TopBar navigation={props.navigation} showMenu={true} showHeader={false} showLogout={true} showUser={true} />,
     });
   };
 
+  public async componentDidMount() {
```

Comment:
async life-cycle methods need error handlers, throw exceptions are just ignored by the react framework